src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -129,9 +229,46 @@ private string GenerateJsonStr(EventWrittenEventArgs eventData)
                 json.Add("RelatedActivityId", eventData.RelatedActivityId);
             }
 
-            // Generate string-representation of JSON. Also remove newlines.
-            string jsonString = json.ToString(Newtonsoft.Json.Formatting.None);
-            jsonString = jsonString.Replace("\n", "\\n").Replace("\r", "\\r");
+            string jsonString = "";
+            if (!this.writeToConsole)
+            {
+                // This path supports the legacy regex-based parser in Linux Dedicated.
+                // In the future, we'll be able to remove this and use JSON logging only
+                IList<string> jsonKeys = json.Properties().Select(p => p.Name).ToList();
+                List<string> regexValues = new List<string>();
+
+                foreach (string column in this.orderedRegexCol)
+                {
+                    string val = "";
+                    if (jsonKeys.Contains(column))
+                    {
+                        try
+                        {
+                            val = (string)json[column];
+                            val = val.Replace("\n", "\\n").Replace("\r", "\\r").Replace("\"", "'").Replace(",", ";;");
+                        }
+                        catch
+                        {
+                        }
+                    }
+
+                    if (column == "Details" || column == "ExceptionMessage")
+                    {
+                        val = '"' + val + '"';
```

Comment:
What's the reason for surrounding these values with quotes? It seems there are some undocumented assumptions here about the expected format of these logs. :)

Reply:
Yeah sorry, this commit you're reviewing is incomplete. I'm going to turn this into a draft PR to avoid getting reviews for in-progress work :)